js/calendar.js

```diff
@@ -1,43 +1,17 @@
-export function renderCalendarCells() {
-    document.addEventListener('DOMContentLoaded', () => {
-        const calendarCells = document.getElementById('calendarCells');
-        const cellsInOneColumn= 19;
-        const daysInWeek = 7;
-        for (let i = 0; i < cellsInOneColumn * daysInWeek; i++) {
-            const cell = document.createElement('div');
-            cell.className = 'cell';
-            calendarCells.appendChild(cell);
-        }
-    });
-}
-function getStartOfWeek(date) {
-    const day = date.getDay(); 
-    const diff = date.getDate() - day + (day === 0 ? -6 : 1); 
-    return new Date(date.setDate(diff));
-}
-function isToday(date) {
-    const today = new Date();
-    return date.getDate() === today.getDate() &&
-           date.getMonth() === today.getMonth() &&
-           date.getFullYear() === today.getFullYear();
-}
-export function renderWeekHeader(dateToRender) {
-    const weekHeader = document.querySelector('.week-header');
-    weekHeader.innerHTML='';
+import { renderWeekHeader } from './header.js';
+import { loadEventsForCurrentWeek } from './events.js';
 
-    const startOfWeek = getStartOfWeek(dateToRender);
-    const wholeWeekHeader = document.createElement('div');
-    wholeWeekHeader.className = 'day'
-    for (let i = 0; i < 7; i++) {
-        const day = new Date(startOfWeek);
-        day.setDate(day.getDate() + i);
-        const dayName = day.toLocaleDateString('en-US', { weekday: 'short' });
-        const dayHeader = document.createElement('div');
-        dayHeader.className = 'day';
-        dayHeader.innerHTML = `<span class='day-name'>${dayName}</span><span class="day-number">${day.getDate()}</span>`;
-        if (isToday(day)) {
-            dayHeader.classList.add('current-day');
-        }
-        weekHeader.appendChild(dayHeader);
-    }
-}
\ No newline at end of file
+export const renderCalendarCells = (date, openModal) => {
+  const calendarCells = document.getElementById("calendarCells");
+  calendarCells.innerHTML = ''; 
+  const cellsInOneColumn = 19;
+  const daysInWeek = 7;
+  for (let i = 0; i < cellsInOneColumn * daysInWeek; i++) {
+    const cell = document.createElement("div");
+    cell.className = "cell";
+    cell.addEventListener('click', () => openModal(date)); 
+    calendarCells.appendChild(cell);
+  }
+  renderWeekHeader(date);
```

Comment:
since you have renderWeekHeader here (which I think is better place), do you still need to renderWeekHeader in header.js?